inst/scripts/make-zilionis-lung-data.Rmd

```diff
@@ -0,0 +1,187 @@
+---
+title: Processing the Zilionis lung cancer dataset
+author: Jens Preussner
+date: "May 4, 2020"
+output:
+  BiocStyle::html_document:
+    titlecaps: false
+    toc_float: true
+bibliography: ref.bib
+---
+
+```{r style, echo=FALSE, results='hide', message=FALSE}
+library(BiocStyle)
+knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
+```
+
+# Downloading the count data
+
+We obtain a single-cell RNA sequencing dataset of human and mouse lung cancer from .
+Counts for endogenous genes are available from the Gene Expression Omnibus
+using the accession number [GSE127465](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE127465).
+We download and cache it using the `r Biocpkg("BiocFileCache")` package.
+
+```{r}
+library(BiocFileCache)
+bfc <- BiocFileCache(ask=FALSE)    
+tarball <- bfcrpath(bfc, 
+    "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE127465&format=file")
+```
+
+We unpack it to a temporary directory.
+
+```{r}
+temp <- tempfile()
+untar(tarball, exdir=temp)
+```
+
+# Reading in human data
+
+We set up a function to load in each set of counts as a sparse matrix.
+
+```{r}
+library(Matrix)
+FUN <- function(X) {
+    input <- read.csv(X, sep="\t", stringsAsFactors=FALSE)
```

Comment:
You can set `row.names=1` to save yourself the need to subset away the first column.